front/pages/w/[wId]/builder/data-sources/[name]/index.tsx

```diff
@@ -344,6 +388,78 @@ function DatasourceDocumentsTabView({
                 className="absolute bottom-8 right-0"
               />
 
+              <>
+                <Dialog
+                  onCancel={() => {
+                    //no-op as we can't cancel file upload
+                  }}
+                  onValidate={() => {
+                    //no-op as we can't cancel file upload
+                  }}
+                  // isSaving is always true since we are showing this Dialog while
+                  // uploading files only
+                  isSaving={true}
+                  isOpen={bulkFilesUploading !== null}
+                  title={`Uploading files`}
+                >
+                  Processing {bulkFilesUploading?.completed}/
+                  {bulkFilesUploading?.total}
+                </Dialog>
+                <input
+                  className="hidden"
+                  type="file"
+                  accept=".txt, .pdf, .md, .csv"
+                  ref={fileInputRef}
+                  multiple={true}
+                  onChange={async (e) => {
+                    if (e.target.files && e.target.files.length > 0) {
+                      const files = e.target.files;
+                      trackMultiFilesUploadUsed({
+                        fileCount: files.length,
+                        workspaceId: owner.sId,
+                      });
+                      let i = 0;
+                      for (const file of files) {
+                        setBulkFilesUploading({
+                          total: files.length,
+                          completed: i++,
+                        });
+                        try {
+                          const res = await handleFileUploadToText(file);
+                          if (res.isOk()) {
+                            await handleUpsert(res.value.content, file.name);
+                          } else {
+                            sendNotification({
+                              type: "error",
+                              title: `Error uploading document ${file.name}`,
+                              description: res.error.message,
+                            });
+                          }
+                        } catch (e) {
+                          sendNotification({
+                            type: "error",
+                            title: "Error uploading document",
+                            description: `An error occurred while uploading your documents.`,
+                          });
+                        }
+                      }
+                      setBulkFilesUploading(null);
+                      router.reload();
```

Comment:
This reload is really not user friendly. We just need to mutate here right?

Reply:
Also in case of errors it prevents the user from reading the error as the notifications go away due to this reload